gsy_myco_sdk/matching_algorithms/preferred_partners_algorithm.py

```diff
@@ -30,85 +31,30 @@ class PreferredPartnersMatchingAlgorithm(BaseMatchingAlgorithm):
 
     Match the bids with their preferred trading partners offers using the PAB matching algorithm.
     A trading partner is a preferable partner that should be matched with.
-    """
-    @classmethod
-    def get_matches_recommendations(
-            cls, matching_data: Dict) -> List[BidOfferMatch.serializable_dict]:
-        recommendations = []
-        for market_id, time_slot_data in matching_data.items():
-            for time_slot, data in time_slot_data.items():
-                if not (data.get("bids") and data.get("offers")):
-                    continue
 
-                recommendations.extend(cls._perform_trading_partners_matching(
-                    market_id, time_slot, bids=data.get("bids"), offers=data.get("offers")))
-        return recommendations
+    This is a variant of the PAB algorithm, it works as following:
+        1. Iterate over bids
+        2. Iterate over requirements of each bid
+        3. Make sure there is a `trading_partners` requirement
+        4. Iterate over trading partners
+        5. Check if there are offers for each partner (from cache {seller_id: [sellers..]})
+        6. Iterate over the requirements of the candidate offer
+        7. Calculate the match's possible selection of energy and clearing rate
+        8. Validate whether the offer/bid can satisfy each other's energy requirements
+        9. Create a match recommendation
+        """
```

Comment:
This could be updated, right?
AFAIS; there is the "new" step to loop over the market slots and also that we re-use the residual bids directly, right?

Reply:
You're right, I'll rephrase this docstring to reflect the added steps